metrics/deploy/prometheus-ocs-rules.yaml

```diff
@@ -404,16 +404,31 @@ spec:
       for: 30m
       labels:
         severity: warning
-    - alert: MDSCPUUsageHigh
+    - alert: MDSCPUUsageHighNeedsHorizontalScaling
+      annotations:
+        description: |-
+          Ceph metadata server pod ({{ $labels.pod }}) has high cpu usage and cannot cope up
+          with the current rate of mds requests.
+          Please consider scaling horizontally by adding more MDS pods.
+        message: Ceph metadata server pod ({{ $labels.pod }}) has high cpu usage
+        runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/openshift-container-storage-operator/CephMdsCpuUsageHighNeedsHorizontalScaling.md
+        severity_level: warning
+      expr: |
+        (label_replace(pod:container_cpu_usage:sum{pod=~"rook-ceph-mds.*"}/ on(pod, namespace) kube_pod_resource_limit{resource='cpu',pod=~"rook-ceph-mds.*"}, "ceph_daemon", "mds.$1", "pod", "rook-ceph-mds-(.*)-(.*)") + on (ceph_daemon, namespace) group_left(managedBy) (0 * (ceph_mds_metadata ==1)) > 0.67) and on (ceph_daemon, namespace, managedBy) (rate(ceph_mds_request[1h]) > 1000)
```

Comment:
High request leading to high cpu usage can be helped with offloading the it to multiple mds, and low requests but still high CPU usage might be because of lack of resources. That's what I understand from the alert

Reply:
@umangachapagain , @weirdwiz , we have made a change where we are not changing the `MDSCPUUsage` alert expression (except for a minor cosmetic change), but we are changing the `description` and `runbook_url` link according to the mds request load.

> High request leading to high cpu usage can be helped with offloading the it to multiple mds, and low requests but still high CPU usage might be because of lack of resources. That's what I understand from the alert

@weirdwiz , yes you are absolutely right.

@umangachapagain ,
>  Why are we comparing with 0.67? 1000?

About the `0.67` not really sure, as this was already there for the existing `MDSCPUUsageHigh` alert, which we never changed. A logical conclusion I draw here is, if you are using more than ≈70%-ntage of CPU for past 6 hours, then it is considered as a sign of high CPU usage.

> Why does >1000 mean HorizontalScaling and <1000 mean VerticalScaling for the same expression?

Now by keeping `67%` as our CPU threshold,  the number `1000` was reached during testing, when (approx) 1000 or more requests were hitting the MD server, we saw a gradual CPU usage rise and in a 1hr window frame it reaches the CPU threshold.
That means if the rate of mds-requests is approx 1000 reqs / sec for an hour we see CPU usage crosses `67%` threshold.

PS: please see the new changes, here we are not doing much modification to the expression, but making `description` and `runbook_url` text changes according to the query (for past 6hrs rate query: `rate(ceph_mds_request[6h])`)